src/com/photoncat/aiproj2/heuristic/Heuristic2.java

```diff
@@ -126,32 +116,32 @@ private int evaluateNByM(Board board, int row, int col, PieceType player) {
             }
             if (flagFirst){
                 if (getByDirLen(board,row,col,i,idx)==0){
-                    score+=3000000;
+                    score+=3000000;     //01111*
                 }
                 else {
-                    score+=25000;
+                    score+=25000;       //21111*
                 }
-                if (player==PieceType.CROSS){
+                if (computerFlag){
                     score+=500;
                 }
                 continue;
             }
             //111*1 or 11*11 etc
-            boolean tierFlag=true;
-            int loopTime=(M+1)/2-1;
+            boolean tierFlag=true;      //flag to indicate whether we find a valid pattern in this level
+            int loopTime=(M+1)/2-1;     // all the position * can be inserted at
             int leftEnd=-M+1;
             int rightEnd=2;
             for (int l=0;l<loopTime;l++){
                 int left=-1;
                 int right=1;
-                boolean flag=true;
+                boolean flag=true;      //flag to indicate whether the pattern is valid
```

Comment:
Consider naming the variable valid instead of "flag" which contains nothing useful